packages/types/src/user-config.ts

```diff
@@ -0,0 +1,104 @@
+import { AgGridExportedName as AgGridExportName } from './exported-name';
+
+export type Framework = 'angular' | 'react' | 'vue' | 'vanilla';
+
+export type ImportType = 'esm' | 'cjs' | 'umd';
+
+export interface MatchGridImportArgs {
+  /** The path being imported, as specified in the source code. For example `import "@my-company/my-grid"` */
+  importPath: string;
+
+  /** The type of module being imported (esm, cjs, umd) */
+  importType: ImportType;
+
+  /**
+   * The framework being matched (vanilla, react, angular, vue).
+   * - @ag-grid-community/angular is 'angular'
```

Comment:
The code mod also works with packages. i.e ag-grid-angular, ag-grid-react, ag-grid-vue, ag-grid-community